src/core/ComposableElement.php

```diff
@@ -115,25 +115,26 @@ protected function init(): void {
     // This function has been micro-optimized
     $new_children = Vector {};
     foreach ($children as $xhp) {
-      /* HH_FIXME[4273] bogus "XHPChild always truthy" - FB T41388073 */
-    if ($xhp instanceof :x:frag) {
-      foreach ($xhp->children as $child) {
-        $new_children->add($child);
-      }
-    } else if (!($xhp instanceof Traversable)) {
-      $new_children->add($xhp);
-    } else {
-      foreach ($xhp as $element) {
-        if ($element instanceof :x:frag) {
-          foreach ($element->children as $child) {
-            $new_children->add($child);
+      if ($xhp is :x:frag) {
```

Comment:
Sorry for messing with the indent. Hackfmt _fixed_ the mismatched indentation level during coding.